src/test/java/core/basesyntax/service/impl/TransactionServiceImplTest.java

```diff
@@ -0,0 +1,124 @@
+package core.basesyntax.service.impl;
+
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationStrategy;
+import core.basesyntax.strategy.BalanceOperationHandler;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.PurchaseOperationHandler;
+import core.basesyntax.strategy.ReturnOperationHandler;
+import core.basesyntax.strategy.SupplyOperationHandler;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class TransactionServiceImplTest {
+    private static final Fruit FRUIT_BANANA = new Fruit("banana");
+    private static final Fruit FRUIT_APPLE = new Fruit("apple");
+    private static final int DEFAULT_QUANTITY = 111;
+    private static final Map<Fruit, Integer> storage = Storage.storage;
+    private static TransactionServiceImpl transactionService;
+    private static Map<FruitTransaction.Operation, OperationHandler> operationHandlerMap;
+    private static OperationStrategy strategy;
+    private static List<FruitTransaction> testTransactionsList;
+
+    @BeforeClass
+    public static void beforeAll() {
+        operationHandlerMap = new HashMap<>();
+        operationHandlerMap.put(FruitTransaction.Operation.BALANCE, new BalanceOperationHandler());
+        operationHandlerMap.put(FruitTransaction.Operation.PURCHASE,
+                new PurchaseOperationHandler());
+        operationHandlerMap.put(FruitTransaction.Operation.RETURN, new ReturnOperationHandler());
+        operationHandlerMap.put(FruitTransaction.Operation.SUPPLY, new SupplyOperationHandler());
+        strategy = new OperationStrategyImpl(operationHandlerMap);
+        transactionService = new TransactionServiceImpl(strategy, new StorageDaoImpl());
+        testTransactionsList = new ArrayList<>();
+    }
+
+    @Test
+    public void applyTransactions_withEmptyTransactionsList_Ok() {
+        storage.put(FRUIT_BANANA, DEFAULT_QUANTITY);
+        int expected = storage.get(FRUIT_BANANA);
+        transactionService.applyTransactions(testTransactionsList);
+        int actual = storage.get(FRUIT_BANANA);
+        assertEquals(expected, actual);
```

Comment:
it would be better to create new map and check that it equals to storage

Reply:
+Corrected in new commit.